pkg/controller/managedcluster/managedcluster_controller.go

```diff
@@ -96,6 +110,26 @@ func (r *ReconcileManagedCluster) Reconcile(ctx context.Context, request reconci
 		return reconcile.Result{}, nil
 	}
 
+	// add a detaching condition to the managed cluster if the managed cluster is deleting
+	// if it is already in detaching or force deaching state, skip it
+	ic := meta.FindStatusCondition(managedCluster.Status.Conditions, constants.ConditionManagedClusterImportSucceeded)
```

Comment:
Do the importing and detaching share the same condition type with different reasons? why not use separated condition types?

Reply:
yes, Importing and detaching represent a state of the life cycle. They have the same dimension and are mutually exclusive. It's easier to use the same type of condition